### DISCUSSION TITLE
contentEditable typescript errors

### DISCUSSION THREAD

**Discussion by nephix** [chakra-ui/zag#1072] [Q&A] [answered]
I'm getting typescript errors in various places with:

```
"@zag-js/react": "^0.31.1",
"@zag-js/switch": "^0.31.1",
```

(it's not limited to `switch`, also occurs in others)

The error:

```
Types of property 'contentEditable' are incompatible.
    Type '"inherit" | Booleanish | undefined' is not assignable to type 'boolean | SignalLike<boolean | undefined> | undefined'.
      Type '"inherit"' is not assignable to type 'boolean | SignalLike<boolean | undefined> | undefined'.ts(2322)
```

```
const [state, send] = useMachine(
    machine({
      id: "1",
    }),
  );
const api = connect(state, send, normalizeProps);

return (
      <label  {...api.rootProps}>
      ...
      </label>
)
```

I saw in https://github.com/chakra-ui/zag/blob/a3eee541c9455691b1ef124dc276b729f0e1229c/packages/types/src/jsx.ts#L593 that `'inherit'` is allowed

In one of my projects I use `preact` (correctly aliased so it works with react stuff) which doesn't seem to allow `inherit`: https://github.com/preactjs/preact/blob/2629e407a5ad9f0a4ae1acf40da1e38a78a10018/src/jsx.d.ts#L2125 

After checking https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/contenteditable I saw that it's either of type `boolean` or `'plaintext-only'`.

Is there any merit to allow `'inherit'`?

I didn't create an issue because I didn't know if it is one

**Comment by segunadebayo** [comment, marked as answer]
The current types we use were inspired by the react typings. Not sure if this has changed since we lasted copied it over.

Feel free to open a PR that updates the typings to the correct version. Let’s see whether they might be type errors across the frameworks
